databricks/sdk/errors/parser.py

```diff
@@ -124,24 +37,43 @@ def _unknown_error(response: requests.Response) -> str:
         f'https://github.com/databricks/databricks-sdk-go/issues. Request log:```{request_log}```')
 
 
-def get_api_error(response: requests.Response) -> Optional[DatabricksError]:
+class _Parser:
     """
-    Handles responses from the REST API and returns a DatabricksError if the response indicates an error.
-    :param response: The response from the REST API.
-    :return: A DatabricksError if the response indicates an error, otherwise None.
+    A parser for errors from the Databricks REST API. It attempts to deserialize an error using a sequence of
+    deserializers, and then customizes the deserialized error using a sequence of customizers. If the error cannot be
+    deserialized, it returns a generic error with debugging information and instructions to report the issue to the SDK
+    issue tracker.
     """
-    if not response.ok:
-        content = response.content
-        for parser in _error_parsers:
-            try:
-                error_args = parser.parse_error(response, content)
-                if error_args:
-                    return _error_mapper(response, error_args)
-            except Exception as e:
-                logging.debug(f'Error parsing response with {parser}, continuing', exc_info=e)
-        return _error_mapper(response, {'message': 'unable to parse response. ' + _unknown_error(response)})
 
-    # Private link failures happen via a redirect to the login page. From a requests-perspective, the request
-    # is successful, but the response is not what we expect. We need to handle this case separately.
-    if _is_private_link_redirect(response):
-        return _get_private_link_validation_error(response.url)
+    def __init__(self,
+                 extra_error_parsers: Optional[List[_ErrorDeserializer]] = None,
+                 extra_error_customizers: Optional[List[_ErrorCustomizer]] = None):
+        self._error_parsers = _error_parsers + (extra_error_parsers
+                                                if extra_error_parsers is not None else [])
+        self._error_customizers = _error_customizers + (extra_error_customizers
+                                                        if extra_error_customizers is not None else [])
+
```

Comment:
When I did this, I get a lint warning that the default argument is mutable. This is probably save since we never update the argument. Will add a comment to skip that lint.

Reply:
I don't think I can skip the lint, and from what I can see, using `None` as default value and then supplying the default in the function for mutable types (list/dict/set) seems to be standard. https://stackoverflow.com/questions/41686829/why-does-pycharm-warn-about-mutable-default-arguments-how-can-i-work-around-the.

Once we drop support for Python 3.8, we can use `|` to express this with fewer braces, which might be easier to read: `list[_ErrorDeserializer] | None`

Reply:
> When I did this, I get a lint warning that the default argument is mutable.

OK I didn't know about that, let's skip it then.